homeassistant/components/conversation/__init__.py

```diff
@@ -393,6 +412,25 @@ def _get_debug_targets(
         yield state, is_matched
 
 
+def _get_unmatched_slots(
+    result: RecognizeResult,
+) -> dict[str, str | int]:
+    """Return a dict of unmatched text/range slot entities."""
+    unmatched_slots: dict[str, str | int] = {}
+    for entity in result.unmatched_entities_list:
+        if isinstance(entity, UnmatchedTextEntity):
+            if entity.text == MISSING_ENTITY:
+                # Don't report <missing> since these are just missing context
+                # slots.
+                continue
+
+            unmatched_slots[entity.name] = entity.text
+        elif isinstance(entity, UnmatchedRangeEntity):
+            unmatched_slots[entity.name] = entity.value
```

Comment:
Let’s cover this in a test

Reply:
Good thing you suggested this. I had a bug :D